src/wasm/wasm-compiler.cc

```diff
@@ -1528,9 +1528,9 @@ Node* WasmGraphBuilder::LoadMem(wasm::LocalType type, MachineType memtype,
 
   *effect = load;
 
-  if (type == wasm::kAstI64 && ElementSizeLog2Of(memtype) < 3) {
+  if (type == wasm::kAstI64 && ElementSizeLog2Of(memtype.representation()) < 3) {
     // TODO(titzer): TF zeroes the upper bits of 64-bit loads for subword sizes.
-    if (TypeOf(memtype) == kTypeInt32) {
+    if (memtype.IsSigned()) {
```

Comment:
I use IsSigned here because I think it is more readable. It should do the same (plus a bit more that does not matter). Here is its implementation:
  bool IsSigned() {
    return semantic() == MachineSemantic::kInt32 ||
           semantic() == MachineSemantic::kInt64;
  }